decrypt_odrive_file.py

```diff
@@ -28,38 +29,89 @@ def derive_key(salt, password):
-        raise ValueError("Invalid Filename: {}".format(ciphertextName))
+    if not padded_plaintext.startswith('\0\0\0\0'):
+        print("Invalid Filename: {}".format(ciphertext_name))
+        return INVALID_NAME
 
-    prefixedName = unpad_pkcs7(paddedPlaintext).decode('utf-8')
-    plaintextName = prefixedName[4:]
+    prefixed_name = unpad_pkcs7(padded_plaintext).decode('utf-8')
+    plaintext_name = prefixed_name[4:]
 
-    return plaintextName
+    return plaintext_name
+
+def single_file(args,file_path):
+    if sys.platform.startswith('win32'):
+        full_file_name = file_path
+    else:
+        full_file_name = file_path
+
+    if (not os.path.isfile(full_file_name) and not os.path.isdir(full_file_name)):
+        print("Error: File/Folder {} not found".format(full_file_name))
+        return
+    elif not full_file_name.endswith(('.cloud', '.cloudf')):
+        decrypted_name = decrypt_name(os.path.basename(full_file_name),args.password) 
+        if decrypted_name != INVALID_NAME:
+            if args.nameonly is False:
+                if (os.path.isdir(full_file_name)):
+                    if args.renamefolder and not os.path.isdir(os.path.join(os.path.dirname(full_file_name),decrypted_name)):
+                        os.rename(full_file_name, os.path.join(os.path.dirname(full_file_name),decrypted_name))
+                        print("'" + full_file_name + "' renamed to '" + decrypted_name + "'")
+                    else:
+                        print("'" + full_file_name + "' not renamed to '" + decrypted_name + "'")
+                else:
+                    if not os.path.isfile(os.path.join(os.path.dirname(full_file_name),decrypted_name)):
+                        with open(full_file_name, 'rb') as in_file, open(os.path.join(os.path.dirname(full_file_name),decrypted_name), 'wb') as out_file:
+                            decrypt_file(in_file, out_file, args.password)
+                        print("Decrypted file written to {}".format(os.path.abspath(out_file.name)))
+                        in_file.close()
+                        out_file.close()
+            else:
+                print os.path.abspath(full_file_name)[4:] + ";" + decrypted_name
+
+def all_files(args, file_path):
+    filesRemain = 1    
+    while filesRemain:
+        filesRemain = 0
+        for root, dirs, files in os.walk(file_path):
+            for f in files:
+                if not f.endswith(('.cloud', '.cloudf')):
+                    decrypted_file_name = decrypt_name(f,args.password)
+                    decrypted_file_path = os.path.join(root,decrypted_file_name) 
+                    if ((decrypted_file_name != INVALID_NAME and not os.path.isfile(decrypted_file_path)) and ((args.filter is None) or (args.filter is not None and args.filter in os.path.join(root,decrypted_file_name)))):
+                        filesRemain = 1
+                        single_file(args,os.path.join(root,f))
```

Comment:
decrypt_name is being called twice for each file - here and in the call to single_file. Maybe part of single_file() should be split out into a separate function:  _decrypt_and_rename(encrypted_path, decrypted_path)

Or maybe performance isn't enough of an issue to get this precise?